compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -830,6 +837,11 @@ public void visit(BLangSimpleVariable varNode) {
                         DiagnosticErrorCode.CONFIGURABLE_VARIABLE_CURRENTLY_NOT_SUPPORTED, lhsType);
             }
         }
+
+        if (varNode.typeNode != null) {
+            analyzeTypeNode(varNode.typeNode, env);
```

Comment:
IMO we should try and call analyzeDef for the type nodes or we aren't quite making use of the visitor pattern. I guess we can move the check for org.wso2.ballerinalang.compiler.tree.types.BLangRecordTypeNode#analyzed into the visit of `BLangRecordTypeNode` itself.

Let's do this separately.

Reply:
Created https://github.com/ballerina-platform/ballerina-lang/issues/30575